nwb_conversion_tools/datainterfaces/axonadatainterface.py

```diff
@@ -0,0 +1,136 @@
+"""Authors: Steffen Buergers"""
+import random
+import re
+import datetime
+import string
+import spikeextractors as se
+
+from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
+
+
+def parse_generic_header(filename, params):
+    """
+    Given a binary file with phrases and line breaks, enters the
+    first word of a phrase as dictionary key and the following
+    string (without linebreaks) as value. Returns the dictionary.
+
+    INPUT
+    filename (str): .set file path and name.
+    params (list or set): parameter names to search for.
+
+    OUTPUT
+    header (dict): dictionary with keys being the parameters that
+                   were found & values being strings of the data.
+
+    EXAMPLE
+    parse_generic_header('myset_file.set', ['experimenter', 'trial_time'])
+    """
+    header = {}
+    params = set(params)
+    with open(filename, 'rb') as f:
+        for bin_line in f:
+            if b'data_start' in bin_line:
+                break
+            line = bin_line.decode('cp1252').replace('\r\n', '').\
+                replace('\r', '').strip()
+            parts = line.split(' ')
+            key = parts[0]
+            if key in params:
+                header[key] = ' '.join(parts[1:])
+
+    return header
+
+
+def read_iso_datetime(set_file):
+    """
+    Creates datetime object (y, m, d, h, m, s) from .set file header
+    """
+    with open(set_file, 'r', encoding='cp1252') as f:
+        for line in f:
+            if line.startswith('trial_date'):
+                date_string = re.findall(r'\d+\s\w+\s\d{4}$', line)[0]
+            if line.startswith('trial_time'):
+                time_string = line[len('trial_time')+1::].replace('\n', '')
+
+    return datetime.datetime.strptime(date_string + ', ' + time_string,
+                                      "%d %b %Y, %H:%M:%S").isoformat()
+
+
+class AxonaRecordingExtractorInterface(BaseRecordingExtractorInterface):
+    """
+    Primary data interface class for converting a AxonaRecordingExtractor
+    """
+
+    RX = se.AxonaRecordingExtractor
+
+    @classmethod
+    def get_source_schema(cls):
+        source_schema = {
+            'required': ['filename'],
+            'properties': {
+                'filename': {
+                    'type': 'string',
+                    'format': 'file',
+                    'description': 'Path to Axona files.'
+                }
+            },
+            'type': 'object',
+            'additionalProperties': True
+        }
+        return source_schema
+
+    def __init__(self, **source_data):
+        super().__init__(**source_data)
+
+    def get_metadata(self):
+        """
+        Auto-fill metadata where possible. Must comply with metadata schema.
+        """
```

Comment:
@sbuergers You can safely remove docstrings for all interface functions unless you are overriding with new/extra arguments that need describing. This will let your interface inherit the hopefully more robust/descriptive (but maybe not quite yet) docstrings from the base class.